src/log.py

```diff
@@ -0,0 +1,44 @@
+import bs4
+import requests
+
+#create a log class, that will parse the log file and return the info
+def parse_log(url):
+    #get the log file
+    r = requests.get(url)
+    #parse the log file
+    soup = bs4.BeautifulSoup(r.text, "html.parser")
+    soup.encode("utf-8")
+    #find the log info
+    data = soup.find("table", {"id": "metadata"})
+    
+    log_info = {
+        "StardewVersion": data.get("data-game-version"),
+        "SMAPI_ver": data.get("data-smapi-version"),
+        "SMAPIMods": data.get("data-code-mods"),
+        "ContentPacks": data.get("data-content-packs"),
+        "OS": data.get("data-os"),
+        "gamepath": data.get("data-game-path"),
+        "success": True,
+    }
+    for key in log_info:
+        if log_info[key] == None:
+            log_info["success"] = False
+    
+    fix_elems = soup.find("ul", {"id": "fix-list"}).find_all("li")
+    fixes = []
+    for fix in fix_elems:
+        innerHTML = fix.decode_contents()
+        if "PyTK 1.23.* or earlier isn't compatible with newer SMAPI performance" in innerHTML:
+            fixes.append("PyTKPerformanceOptimizationsOff")
```

Comment:
"PyTKPerformanceOptimizationsOff" is the text we'll be showing the users right? We probably want that to be more verbose; the average SMAPI user will have no idea what that means (same for the others).

Reply:
true true